### DISCUSSION TITLE
Managing Ansible inventory with CUE

### DISCUSSION THREAD

**Discussion by ryancraig** [cue-lang/cue#2056] [Show and tell]
I'm working on a PoC to showcase the benefits of managing Ansible inventory source with CUE. Once I have a working implementation, I plan to publish an article on Medium. https://medium.com/@ryan.s.craig/schema-validated-and-policy-driven-ansible-inventory-management-cue-it-up-f970fe54a6a2

Before I publish my article, I'll post a draft link to this discussion. I'd appreciate comments and critiques. I think this use case will introduce CUE to a large audience including many enterprise users.

**Comment by trantor** [comment]
I had started handling (partially) my host_vars YAML files using CUE in order to validate them. I have not reached the point where I decide how much of ansible multiple-level overrides/inheritance for host/group variables to keep, since the approach of CUE goes in quite a different (if not opposite) direction.
At some point I'll manage to find the time to go back to the issue, but any experience from others would be very interesting.

**Reply by ryancraig** [reply]
Agreed, using CUE will not facilitate the use of Ansible's default order of variable override. In large organizations where inventory development is distributed amongst many teams, Ansible's variable override flexibility seems to become a liability quickly. In my experience, enforcing schema and policy will be valued more than achieving perfect DRYness in K/Vs. My hope is that I can reduce the failures we constantly experience caused by inventory errors. It'll definitely be an interesting experiment.

**Comment by David2011Hernandez** [comment]
Hej, I think this could be an interesting idea, I would like to spent some time fiddling with it, Do you have any updates on this?
